Source/WebCore/rendering/RenderButton.cpp

```diff
@@ -169,4 +170,10 @@ void RenderButton::layout()
 }
 #endif
 
+// For compatibility with IE/FF we only clip overflow on input elements.
+bool RenderButton::hasControlClip() const
+{
+    return !is<HTMLButtonElement>(formControlElement());
```

Comment:
I think we should use `is<HTMLInputElement>` to avoid unnecessary negation, and better align with the comment.